Get ranges.py

```diff
@@ -0,0 +1,72 @@
+"""
```

Comment:
too complex

Reply:
I agree, for a better solution I need to spend a lot of time ... 
The solution is working - and this is the main thing ...

Reply:
no need in the working code if you can't change it after a half-year delay.

Reply:
also O(n**2) complexity